src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -398,10 +398,15 @@ function updateDOMProperties(
     } else if (propKey === DANGEROUSLY_SET_INNER_HTML) {
       var nextHtml = nextProp ? nextProp[HTML] : undefined;
       var lastHtml = lastProp ? lastProp[HTML] : undefined;
-      if (nextHtml) {
+      if (
+        nextHtml ||
```

Comment:
This is for consistent with stack reconciler.
Because stack reconciler can render other than string and number.

* http://codepen.io/anon/pen/BQXwVP?editors=0010